Add MQTT BBEs (#4829)
ballerina-platform/ballerina-distribution: examples/mqtt-service-subscribe-message/mqtt_service_subscribe_message.md

# MQTT service - Subscribe to messages

> The `mqtt:Service` connects to a given MQTT server via the `mqtt:Listener`, and allows to receive messages from different topics. Once subscribed to a topic, when new messages are received, the `onMessage` method gets invoked. If there are errors when invoking the method, the `onError` remote method will be invoked with the relevant `error`. If the method is not implemented, the error will be logged to the console. Use this to receive messages from a set of topics in an MQTT server.

Reviewer: ```suggestion
The `mqtt:Service` connects to a given MQTT server via the `mqtt:Listener`, and allows to receive messages from different topics. Once new messages are received from a subscribed topic, the `onMessage` method gets invoked. If there are errors when invoking the method, the `onError` remote method will be invoked with the relevant `error`. If the method is not implemented, the error will be logged to the console. This can be used to receive messages from a set of topics in an MQTT server.
```

::: code mqtt_service_subscribe_message.bal :::

> ## Prerequisites

Reviewer: ```suggestion
## Prerequisites
```
Ditto

- Start a [MQTT broker](https://mqtt.org/software/) instance.

> Run the program by executing the following command.

Reviewer: ```suggestion
Run the program by executing the following command.
```
Ditto

::: out mqtt_service_subscribe_message.out :::

> >**Tip:** Run the MQTT client given in the [MQTT client - Publish message](/learn/by-example/mqtt-client-publish-message) example to publish some messages to the topic.

Reviewer: ```suggestion
>**Tip:** Run the MQTT client given in the [MQTT client - Publish message](/learn/by-example/mqtt-client-publish-message) example to publish some messages to the topic.
```
Ditto
